### PR TITLE
Akinator

### PR DESCRIPTION
**Pull request by NeginsCode**
Do you want to play 20 Questions with Roboy using the Akinator API????? :man_with_turban:

- Trigger ravestate_akinator with: "I like games", "I want to play" or something similar :8ball:
- nlp:yesno detects yes, no, probably ... :bell:
- Fully functioning Akinator library :books:
- Ravestate states handles your individual game experience using a bunch of nice states :cookie:
- Use detached signals (credit to @josephbirkner) :koala:

                                 <\              _
                                  \\          _/{
                           _       \\       _-   -_
                         /{        / `\   _-     - -_
                       _~  =      ( @  \ -        -  -_
                     _- -   ~-_   \( =\ \           -  -_
                   _~  -       ~_ | 1 :\ \      _-~-_ -  -_
                 _-   -          ~  |V: \ \  _-~     ~-_-  -_
              _-~   -            /  | :  \ \            ~-_- -_
           _-~    -   _.._      {   | : _-``               ~- _-_
        _-~   -__..--~    ~-_  {   : \:}
      =~__.--~~              ~-_\  :  /
                                 \ : /__
                                //`Y'--\\
                               <+       \\

### CODE REVIEW

File: modules/ravestate_akinator/api.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+# put in config
+NEW_SESSION_URL = "https://srv11.akinator.com:9152/ws/new_session?callback=&partner=&player=website-desktop&uid_ext_session=&frontaddr=NDYuMTA1LjExMC40NQ==&constraint=ETAT<>'AV'"
+ANSWER_URL = "https://srv11.akinator.com:9152/ws/answer"
+GET_GUESS_URL = "https://srv11.akinator.com:9152/ws/list"
+CHOICE_URL = "https://srv11.akinator.com:9152/ws/choice"
+EXCLUSION_URL = "https://srv11.akinator.com:9152/ws/exclusion"
+
+
+class Api:
```
Comment (Toseban) [inline modules/ravestate_akinator/api.py]: Nice wrapper for the API calls :+1:

File: modules/ravestate_nlp/__init__.py [hunk 2]
```diff
@@ -62,6 +70,14 @@ def nlp_preprocess(ctx):
     ctx["nlp:roboy"] = nlp_roboy
     logger.info(f"[NLP:roboy]: {nlp_roboy}")
 
+    nlp_yesno = nlp_doc._.yesno
+    ctx["nlp:yesno"] = nlp_yesno
+    logger.info(f"[NLP:yesno]: {nlp_yesno}")
+
+    if nlp_triples[0].match_either_lemma(pred={"play"}, obj={"game"}):
```
Comment (josephbirkner) [inline modules/ravestate_nlp/__init__.py]: Why is this a property and not an nlp:intent-play signal or so?

File: modules/ravestate_nlp/yes_no.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+def yes_no(doc):
+    """
+    checks input for "yes", "no", "i don't know", "probably" and "probably not"
+    """
+    nlp_tokens = tuple(str(token) for token in doc)
+    for token in nlp_tokens:
+        if token == "yes" or token == "y":
+            return "yes"
+        elif token == "no" or token == "n":
```
Comment (Toseban) [inline modules/ravestate_nlp/yes_no.py]: maybe check case insensitive
But that might be a change to do somewhere higher up the nlp chain :wink:

Comment (NeginsCode) [reply modules/ravestate_nlp/yes_no.py]: Done :) Thanks

File: modules/ravestate_akinator/api.py (hunk 1 above)
Comment (NeginsCode) [reply modules/ravestate_akinator/api.py]: :relaxed:

File: modules/ravestate_akinator/__init__.py [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+from ravestate_akinator.api import Api
+
+from reggol import get_logger
+logger = get_logger(__name__)
+
+akinator_api: Api
+CERTAINTY = "certainty_percentage"
+
+
+# TODO: Change this to cond=idle:bored
```
[the remark this answers is not in the document]
Comment (NeginsCode) [reply modules/ravestate_akinator/__init__.py]: Done :wave:

File: modules/ravestate_nlp/__init__.py (hunk 2 above)
Comment (NeginsCode) [reply modules/ravestate_nlp/__init__.py]: Changed it :sheep:

**Pull request merged**